### DISCUSSION TITLE
CBOR in Byron transactions

### DISCUSSION THREAD

**Discussion by mkazlauskas** [CardanoSolutions/ogmios#338] [Ideas / Feature Requests]
### What is your idea? Provide a use case.

Either always include `cbor` in Byron transactions, or compute and set the `fee`

### Why is it a good idea?

`fee` is not part of a Byron transaction body and we used cbor it to compute it.

### What is the current alternative and why is it not good enough?

https://github.com/CardanoSolutions/ogmios/commit/c7b686989eca7750934739747b8471d5989cbf54 changed the behavior that was there before 6.0

**Comment by KtorZ** [comment]
I like the idea but it is going to be challenging to know what were the protocol parameters needed to compute fees at the moment of the serialization. 

Perhaps instead including a size in bytes would solve your issue?

**Reply by mkazlauskas** [reply]
Yeah, size in bytes would work!

**Reply by mkazlauskas** [reply]
Do you know if fee coefficient or constant ever changed on mainnet during Byron?

**Reply by KtorZ** [reply]
They haven't. My concern is between mainnet and other networks.

**Reply by KtorZ** [reply]
**Mainnet**

https://github.com/input-output-hk/cardano-configurations/blob/8a403bb2fdb23d17a25fcc14141395918b0e5be0/network/mainnet/genesis/byron.json#L14523-L14525

**PreProd**
https://github.com/input-output-hk/cardano-configurations/blob/8a403bb2fdb23d17a25fcc14141395918b0e5be0/network/preprod/genesis/byron.json#L111-L112

**Preview**

N/A (there's no Byron era on Preview)

**Reply by KtorZ** [reply]
It is tempting.

**Reply by mkazlauskas** [reply]
Probably best to leave it up to applications to compute :sweat_smile:

**Reply by KtorZ** [reply]
Actually, nevermind. It's not even _computable_ for Byron without knowing the inputs! One can compute the _minimum required fee_ from the size, but it isn't necessarily equal to the actual transaction fee. Often, wallets would leave more.

**Reply by mkazlauskas** [reply]
Good point :pensive: It means that we wouldn't actually benefit from knowing transaction size, inputs and outputs is enough. We'll have to update our application to resolve the inputs during the mapping...
